src/k8s/pkg/k8sd/api/cluster_config.go

```diff
@@ -27,12 +27,19 @@ func (e *Endpoints) putClusterConfig(s *state.State, r *http.Request) response.R
 		return response.BadRequest(fmt.Errorf("failed to decode request: %w", err))
 	}
 
+	// Changing the datastore configuration is opt-in. We expect the caller to explicitly set the "external" type.
+	// The nil check is required to ensure we only fail if the DatastoreConfig is expected to change.
+	if req.DatastoreConfig.Type != nil && req.DatastoreConfig.GetType() != "external" {
```

Comment:
i would make this part of `types.DatastoreConfigFromUserFacing` and return a `(Datastore, error)`